src/Common/src/Interop/Windows/kernel32/Interop.WaitForSingleObjectDontcallThis.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+internal partial class Interop
+{
+    internal partial class Kernel32
+    {
+        [DllImport(Libraries.Kernel32, ExactSpelling=true, SetLastError=true, EntryPoint="WaitForSingleObject")]
+        internal static extern int WaitForSingleObjectDontCallThis(SafeWaitHandle handle, int timeout);
```

Comment:
I would rename the file to `Interop.WaitForSingleObject.cs`. And the method name suffix sounds interesting. Why shouldn't it be called?

Reply:
Judging by all the comments in SharedUtils.cs, it's because it's wrapped in special magic to make ConstrainedExecutionRegions work safely. We don't have those CER's any more, so the name is not needed. Please rename it to WaitForSingleObject, or move it out of Common back into perf counters.

Although I also notice that it's potentially dangerous because if it's called on a UI or STA thread, and what it is waiting on causes any Window message to get posted to the thread, the process will deadlock, I think. Even though have a 500ms timeout on the wait, it just does Sleep(0) and tries again. I'm not sure there is any opportunity to pump messages in that loop. Definitely not something to attempt to change now.